app/src/main/java/com/github/se/eventradar/model/EventsOverviewViewModel.kt

```diff
@@ -60,36 +38,102 @@ class EventsOverviewViewModel(db: FirebaseFirestore = Firebase.firestore) : View
         }
   }
 
-  private val userRef = db.collection("users")
-
-  private fun getEventContact(contactId: String): String {
-    var contactEmail = ""
-
+  fun getUpcomingEvents(uid: String) {
+    // Fetch the user's data to get their eventsAttendeeList
     userRef
-        .document(contactId)
-        .collection("private")
-        .limit(1) // Limit the query to retrieve only one document
+        .document(uid)
         .get()
-        .addOnSuccessListener { querySnapshot ->
-          for (documentSnapshot in querySnapshot.documents) {
-            contactEmail = documentSnapshot.data?.get("email") as String
-            break
+        .addOnSuccessListener { userSnapshot ->
+          val user = userFromDocument(userSnapshot)
+          user?.let {
+            val attendeeList = it.eventsAttendeeList
+            if (attendeeList.isNotEmpty()) {
+              eventRef
+                  .whereIn(FieldPath.documentId(), attendeeList)
+                  .get()
+                  .addOnSuccessListener { eventsQuerySnapshot ->
+                    val upcomingEvents =
+                        eventsQuerySnapshot.documents.mapNotNull { eventSnapshot ->
+                          eventFromDocument(eventSnapshot)
+                        }
+                    _uiState.value =
+                        _uiState.value.copy(
+                            eventList =
+                                EventList(
+                                    upcomingEvents,
+                                    upcomingEvents,
+                                    _uiState.value.eventList.selectedEvent))
+                  }
+                  .addOnFailureListener { e ->
+                    Log.d("EventsOverviewViewModel", "Error getting upcoming events: ", e)
+                  }
+            } else {
+              // User has no upcoming events
+              _uiState.value =
+                  _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
+            }
           }
+              ?: run {
+                Log.d("EventsOverviewViewModel", "Could not parse User from DocumentSnapshot.")
+              }
         }
-        .addOnFailureListener { exception ->
-          Log.d("EventsOverviewViewModel", "Error getting event contact: ", exception)
+        .addOnFailureListener { e ->
+          Log.d("EventsOverviewViewModel", "Error fetching user document: ", e)
         }
-
-    return contactEmail
   }
 
   companion object {
-    fun getLocalDateTime(dateTime: String): LocalDateTime {
+    fun eventFromDocument(document: DocumentSnapshot): Event? {
+      val data = document.data ?: return null
+      return Event(
+          eventName = data["name"] as? String ?: "Unknown Event",
```

Comment:
nit: I would much rather prefer having some type of error handling instead of default values. If there are default values then we might not catch if there are problems with the database.